File: src/main/java/me/zeroeightsix/kami/util/ClassFinder.java
package me.zeroeightsix.kami.util;

import org.reflections.Reflections;

import java.util.Set;

/**
 * Created by 086 on 23/08/2017.
 */
public class ClassFinder {

    public static Set<Class> findClasses(String pack, Class subType) {
        Reflections reflections = new Reflections(pack);
        return reflections.getSubTypesOf(subType);
    }
}

File: src/main/java/me/zeroeightsix/kami/util/ClassFinder.kt
package me.zeroeightsix.kami.util

import org.reflections.Reflections

/**
 * Created by 086 on 23/08/2017.
 */
object ClassFinder {
    @JvmStatic
    fun <T> findClasses(pack: String?, subType: Class<T>?): Array<Class<out T>> {
        val reflections = Reflections(pack)
        return reflections.getSubTypesOf(subType).toTypedArray()
    }
}